File: src/bs_gstl/src/calc_mean.cpp
/*
   Copyright 2009 HPGL Team
   This file is part of HPGL (High Perfomance Geostatistics Library).
   HPGL is free software: you can redistribute it and/or modify it under the terms of the BSD License.
   You should have received a copy of the BSD License along with HPGL.

*/


#include "stdafx.h"
#include <calc_mean.h>
#include <property_array.h>


	double hpgl::calc_mean(const cont_property_array_t & property, bool * success)
	{	
		int count_points = 0;
		double sum = 0;

		// Mean computation procedure
		for(int idx = 0; idx < property.size(); ++idx)
		{
			if( property.is_informed(idx) )
			{
				sum += property.get_at(idx);
				count_points += 1;
			}
		}		
		if (count_points == 0)
		{
			*success = false;
			return 0;
		}
		else
		{
			*success = true;
			return sum / count_points;
		}
	}

namespace hpgl
{

	bool calc_mean(const cont_property_array_t * property, double * mean)
	{		
		if (!property || ! mean)
			return false;

		int count_points = 0;
		double sum = 0;
		// Mean computation procedure
		for(int idx = 0; idx < property->size(); ++idx)
		{
			if( property->is_informed(idx) )
			{
				sum += property->get_at(idx);
				count_points += 1;
			}
		}		
		*mean = sum / count_points;		
		return true;
	}
}

File: src/sample-scripts/calc_mean.py
#
#   Copyright 2009 HPGL Team
#   This file is part of HPGL (High Perfomance Geostatistics Library).
#   HPGL is free software: you can redistribute it and/or modify it under the terms of the BSD License.
#   You should have received a copy of the BSD License along with HPGL.
#

#!/usr/bin/python

from numpy import *
from scipy import *
from pylab import *

if len(sys.argv) <> 2:
	print "Usage: calc_mean.py <file>"
	exit()

filename = sys.argv[1]

values = []
f = open(filename)
for line in f:
	ss = line.split()
	for s in ss:
		try:
			values += [float(s.strip())]
		except:
			pass

sum = 0
count = 0
for v in values:
	if v <> -99:
		sum += v
		count += 1
print sum
print count
print sum/count	

		
